packages/components/src/internal/components/security/APIWrapper.ts

```diff
@@ -184,12 +189,36 @@ export class ServerSecurityAPIWrapper implements SecurityAPIWrapper {
 
     fetchContainers = fetchContainers;
 
-    fetchGroups = (projectPath: string): Promise<FetchedGroup[]> => {
+    fetchGroups = (
+        projectPath: string,
+        permissions?: string | string[],
+        checkIsAdmin?: boolean,
+        permissionCheck?: 'all' | 'any'
+    ): Promise<FetchedGroup[]> => {
         return new Promise((resolve, reject) => {
             Security.getGroupPermissions({
                 containerPath: projectPath,
                 success: data => {
-                    resolve(data?.container?.groups);
+                    const groups = data?.container?.groups;
+                    if (!permissions) resolve(groups);
+
+                    const perms = typeof permissions === 'string' ? [permissions] : permissions;
+                    const groupsWithPerm = groups?.filter(group => {
+                        if (checkIsAdmin && group.id === -1) {
+                            return perms?.length > 0;
+                        } else if (perms) {
+                            const allPerms = group.effectivePermissions ?? [];
+
+                            if (permissionCheck === 'any') {
+                                return perms.some(p => allPerms.indexOf(p) > -1);
+                            } else {
+                                return perms.every(p => allPerms.indexOf(p) > -1);
+                            }
+                        }
+
+                        return false;
```

Comment:
this is so similar to the `hasPermissions()` function we have in ui-components. Is there any way that could be refactored or shared / used here in some way?